internal/literals/numbers.go

```diff
@@ -109,14 +110,14 @@ func obfuscateNumberLiteral(cursor *astutil.Cursor, info *types.Info, obfLits ob
 		return false, nil
 	}
 
-	intValue, err := strconv.ParseInt(strValue, 0, 64)
+	intValue, err := strconv.ParseInt(strings.Split(strValue, ".")[0], 0, 64)
```

Comment:
I honestly don't understand this, still. If this was actually an integer, there shouldn't be a decimal point. If it's any numerical value, then we need to use the right parsing function depending on the kind - ParseInt, ParseUint, ParseFloat, ParseComplex, and so on.

Reply:
`var x int = 1.0` is valid go

Reply:
right, but why should that matter? we are obfuscating the literal, not the variable. the "kind" of the literal is token.FLOAT, so we should just be using ParseFloat on the string value.

If we convert all float32/float64 values to int64 later for obfuscation, that's fine, but it should happen independently of parsing.